### DISCUSSION TITLE
Common code execution with workers

### DISCUSSION THREAD

**Discussion by AndreuCodina** [fastapi/fastapi#11906] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# configuration_container.py
class ConfigurationContainer:
  @classmethod
  def initialize(cls) -> None:
    cls.configurations = cls._read_configurations()

  @classmethod
  def _read_configurations(cls):
    # Read from Azure
    ...


# main.py
ConfigurationContainer.initialize()
```

### Description

When our `main.py` starts, we read configurations (logging level, URLs, passwords, feature flags, etc.) from Azure and store them in a class instance (a singleton). Every read costs money and takes time.

We run FastAPI with N workers in production, and we want to share the singleton between workers. Is this possible?

Apart from that, to not re-redeploy our microservices after every configuration change, we want to implement the initial reading, but every 5 minutes in background. Again we have the same problem: we don't want to read the configurations once for each worker.

### Operating System

Linux

### FastAPI Version

0.111.1

### Pydantic Version

2.8.2

### Python Version

3.12.4

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
